test/test_out_s3.rb

```diff
@@ -689,22 +687,22 @@ def test_warn_for_delay
   end
 
   def test_assume_role_credentials_fail
-    setup_mocks
-    mock(Aws::AssumeRoleCredentials).new(
-      role_arn: "test_arn",
-      role_session_name: "test_session",
-      client: anything
-    ) { raise Aws::STS::Errors::AccessDenied.new("a", "b") }
+    expected_credentials = Aws::Credentials.new("test_key", "test_secret")
+    any_instance_of(Aws::STS::Client) do |klass|
+      stub(klass).assume_role(role_arn: "test_arn", role_session_name: "test_session") {
+        raise Aws::STS::Errors::AccessDenied.new("error", "message")
+      }
+    end
     config = CONFIG_TIME_SLICE.split("\n").reject{|x| x =~ /.+aws_.+/}.join("\n")
     config += %[
       <assume_role_credentials>
         role_arn test_arn
         role_session_name test_session
       </assume_role_credentials>
     ]
-    d = create_time_sliced_driver(config)
-    d.run {}
-    # should not raise an exception
+    d = create_driver(config)
```

Comment:
The difference between driver and timesliced-driver isn’t immediately obvious to me. I’m guessing it changes what we’re testing somehow?

Reply:
The timesliced-driver will output files that have the write time encoded in the file path. For our purposes, the two are interchangeable, as we're testing the assume-role and authentication logic, which is independent.